site.yml

```diff
@@ -192,11 +192,21 @@
 - name: keystone setup
   hosts: controller[0]
   any_errors_fatal: true
+  pre_tasks:
+    # These pre-tasks determine if keystone has been setup before. If so, the
+    # keystone-setup role tasks are skipped.
+    - name: discover keystone setup status
+      command: mysql --batch --silent keystone -e "select type from service where type = 'compute'"
+      failed_when: false
```

Comment:
Lets add a `changed_when: False` too. It wasn't part of the original, but this shouldn't result in a change.